packages/accounts-base/localstorage_token.js

```diff
@@ -8,178 +8,179 @@ var Ap = AccountsClient.prototype;
-Ap._isolateLoginTokenForTest = function () {
-  this.LOGIN_TOKEN_KEY = this.LOGIN_TOKEN_KEY + Random.id();
-  this.USER_ID_KEY = this.USER_ID_KEY + Random.id();
-};
-
-Ap._storeLoginToken = function (userId, token, tokenExpires) {
-  Meteor._localStorage.setItem(this.USER_ID_KEY, userId);
-  Meteor._localStorage.setItem(this.LOGIN_TOKEN_KEY, token);
-  if (! tokenExpires)
-    tokenExpires = this._tokenExpiration(new Date());
-  Meteor._localStorage.setItem(this.LOGIN_TOKEN_EXPIRES_KEY, tokenExpires);
-
-  // to ensure that the localstorage poller doesn't end up trying to
-  // connect a second time
-  this._lastLoginTokenWhenPolled = token;
-};
-
-Ap._unstoreLoginToken = function () {
-  Meteor._localStorage.removeItem(this.USER_ID_KEY);
-  Meteor._localStorage.removeItem(this.LOGIN_TOKEN_KEY);
-  Meteor._localStorage.removeItem(this.LOGIN_TOKEN_EXPIRES_KEY);
-
-  // to ensure that the localstorage poller doesn't end up trying to
-  // connect a second time
-  this._lastLoginTokenWhenPolled = null;
-};
-
-// This is private, but it is exported for now because it is used by a
-// test in accounts-password.
-//
-Ap._storedLoginToken = function () {
-  return Meteor._localStorage.getItem(this.LOGIN_TOKEN_KEY);
-};
-
-Ap._storedLoginTokenExpires = function () {
-  return Meteor._localStorage.getItem(this.LOGIN_TOKEN_EXPIRES_KEY);
-};
-
-Ap._storedUserId = function () {
-  return Meteor._localStorage.getItem(this.USER_ID_KEY);
-};
-
-Ap._unstoreLoginTokenIfExpiresSoon = function () {
-  var tokenExpires = this._storedLoginTokenExpires();
-  if (tokenExpires && this._tokenExpiresSoon(new Date(tokenExpires))) {
-    this._unstoreLoginToken();
-  }
-};
-
-///
-/// AUTO-LOGIN
-///
-
-Ap._initLocalStorage = function () {
-  var self = this;
-
-  // Key names to use in localStorage
-  self.LOGIN_TOKEN_KEY = "Meteor.loginToken";
-  self.LOGIN_TOKEN_EXPIRES_KEY = "Meteor.loginTokenExpires";
-  self.USER_ID_KEY = "Meteor.userId";
-
-  var rootUrlPathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
-  if (rootUrlPathPrefix || this.connection !== Meteor.connection) {
-    // We want to keep using the same keys for existing apps that do not
-    // set a custom ROOT_URL_PATH_PREFIX, so that most users will not have
-    // to log in again after an app updates to a version of Meteor that
-    // contains this code, but it's generally preferable to namespace the
-    // keys so that connections from distinct apps to distinct DDP URLs
-    // will be distinct in Meteor._localStorage.
-    var namespace = ":" + this.connection._stream.rawUrl;
-    if (rootUrlPathPrefix) {
-      namespace += ":" + rootUrlPathPrefix;
+Meteor.loginWithToken = (token, callback) => Accounts.loginWithToken(token, callback);
+
+// XXX is this style of assigning methods OK?
+// refs: http://raganwald.com/2015/06/10/mixins.html
+// https://github.com/meteor/meteor/blob/devel/packages/underscore/underscore.js#L1287
+
+Object.assign(AccountsClient.prototype, {
```

Comment:
Instead of changing the way we augment `AccountsClient.prototype`, perhaps now would be a good time to move these methods into `accounts_client.js` as part of the `AccountsClient` class body.